### PR TITLE
Burning toxens decreases total supply

### LINKED ISSUE #196: Bank Module: Burn tokens should decrease total supply

### PR DESCRIPTION
**Pull request by citizen-stig**
Resolves #196

### CODE REVIEW

File: sov-modules/sov-modules-impl/bank/src/token.rs [hunk 1]
```diff
@@ -52,13 +52,20 @@ impl<C: sov_modules_api::Context> Token<C> {
...
         from: &C::Address,
-        burn: &C::Address,
         amount: Amount,
         working_set: &mut WorkingSet<C::Storage>,
     ) -> Result<CallResponse> {
-        self.transfer(from, burn, amount, working_set)
+        let balance = self.balances.get_or_err(from, working_set)?;
+        let new_balance = match balance.checked_sub(amount) {
+            Some(from_balance) => from_balance,
+            // TODO: Add `from` address to the message (we need pretty print for Address first)
```
Comment (preston-evans98) [inline sov-modules/sov-modules-impl/bank/src/token.rs:63]: We have `AddressBech32` - we can convert on the fly here

File: sov-modules/sov-modules-impl/bank/tests/burn_test.rs [hunk 2]
```diff
@@ -81,10 +81,11 @@ fn burn_deployed_tokens() {
 
     let current_total_supply = query_total_supply(&mut working_set);
     // Total supply does not change
```
Comment (preston-evans98) [inline sov-modules/sov-modules-impl/bank/tests/burn_test.rs:83]: nit: I think this comment is outdated now

**Review by bkolad — APPROVED** [review]

File: sov-modules/sov-modules-impl/bank/tests/burn_test.rs (hunk 2 above)
Comment (citizen-stig) [reply sov-modules/sov-modules-impl/bank/tests/burn_test.rs]: Fixed

File: sov-modules/sov-modules-api/src/lib.rs [hunk 3]
```diff
@@ -119,7 +119,8 @@ pub trait Spec {
         + Into<AddressBech32>
         + Serialize
         + for<'a> Deserialize<'a>
-        + Send;
+        + Send
+        + Display;
```
Comment (citizen-stig) [inline sov-modules/sov-modules-api/src/lib.rs:123]: @bkolad @preston-evans98 
Latest addition so addressed can be printed in message. This is change in API so I want ack for this change.

**Pull request merged**